include/mbgl/map/map.hpp

```diff
@@ -156,6 +156,21 @@ class Map : private util::noncopyable {
 
     AnnotationIDs queryPointAnnotations(const ScreenBox&);
 
+    // Tile prefetching
+    //
+    // When loading a map area, if `FixedPrefetch` is set to any positive number, the map will
+    // request first the tile for the prefetch `zoom` level if the `zoom` level is within the
+    // tile source range and if it is lower than the current zoom level, in a attempt to display
+    // a map as quick as possible. `DynamicPrefetch` works similarly, but will prefetch a tile
+    // at `zoom = (getZoom() - delta)`, constrained at the lowest zoom level for the source.
+    //
+    // If both are set, only a tile for the fixed zoom is requested.
+    void setFixedPrefetchZoom(optional<uint8_t> zoom);
+    optional<uint8_t> getFixedPrefetchZoom() const;
+
+    void setDynamicPrefetchZoomDelta(optional<uint8_t> delta);
+    optional<uint8_t> getDynamicPrefetchZoomDelta() const;
```

Comment:
The fixed is if you want to be cheap: you can set the prefetch zoom level to say, zoom level 5 and you will never have a blank spot on the map, but it will look very low res and overscaled. Soon you will get the whole world at zoom 5 in your cache.

The dynamic IMO is more interesting, but it will use more resources. You could set it to say, 2, and it will always prefetch "CurrentZoom - 2", it will pretty much make you never see blank spots with better looking overscaled tile but it will fill the cache faster and also use more bandwidth.

I find both approaches useful. We could drop the "dynamic" API and leave it to be implemented client side if you think we don't need 2 APIs.